private/pkg/storage/bucket.go

```diff
@@ -252,3 +290,29 @@ type nopReadWriteBucketCloser struct {
 func (nopReadWriteBucketCloser) Close() error {
 	return nil
 }
+
+type putOptions struct {
+	atomic             bool
+	suggestedChunkSize *int
+}
+
+func newPutOptions() *putOptions {
+	return &putOptions{}
+}
+
+func (p *putOptions) Atomic() bool {
+	return p.atomic
+}
+
+func (p *putOptions) SuggestedDisableChunking() bool {
+	return p.suggestedChunkSize != nil && *p.suggestedChunkSize == 0
```

Comment:
If we don't prevent negative values here we should change the conditional from == to <= here.

Reply:
Done